subsys/net/lib/aws_fota/src/aws_fota.c

```diff
@@ -0,0 +1,509 @@
+				"meaning that the current device job was canceled");
+			return 1;
+		}
+		printk("recived err: %d\n", err);
+
+		/* Unsubscribe from notify_next_topic to not recive more jobs
+		 * while processing the current job.
+		 */
+		err = aws_jobs_unsubscribe_notify_next(client);
+		if (err) {
+			LOG_ERR("Error when unsubscribing notify_next_topic: %d",
+				err);
+			return err;
+		}
+
+		/* Subscribe to update topic to recive feedback on wether an
+		 * update is accepted or not.
+		 */
+		err = aws_jobs_subscribe_job_id_update(client, job_id);
+		if (err) {
+			LOG_ERR("Error when subscribing job_id_update: %d", err);
+			return err;
+		}
+
+		/* Construct job_id topics to be used for filtering publish
+		 * messages.
+		 */
+		err = construct_job_id_update_topic(client->client_id.utf8,
+			job_id, "/accepted", job_id_update_accepted_topic);
+		if (err) {
+			LOG_ERR("Error when constructing_job_id_update_accepted: %d",
+				err);
+			return err;
+		}
+
+		err = construct_job_id_update_topic(client->client_id.utf8,
+			job_id, "/rejected", job_id_update_rejected_topic);
+		if (err) {
+			LOG_ERR("Error when constructing_job_id_update_rejected: %d",
+				err);
+			return err;
+		}
+
+		/* Set fota_state to DOWNLOAD_FIRMWARE, when we are subscribed
+		 * to job_id topics we will try to publish and if accepted we
+		 * can start the download
+		 */
+		fota_state = DOWNLOAD_FIRMWARE;
+
+		/* Handled by the library*/
+		return 1;
+
+	} else if (!strncmp(job_id_update_accepted_topic, topic,
+		    MIN(JOB_ID_UPDATE_TOPIC_MAX_LEN, topic_len))) {
+
+		LOG_DBG("Job document update was accepted");
+		err = publish_get_payload(client, payload_len);
+		if (err) {
+			return err;
+		}
+		/* Increment document version counter as the update was accepted */
+		doc_version_number++;
+
+		if (fota_state == DOWNLOAD_FIRMWARE) {
+			/* Job document is updated and we are ready to download
+			 * the firmware. */
+			execution_state = AWS_JOBS_IN_PROGRESS;
+			LOG_INF("Start downloading firmware from %s%s",
+				log_strdup(hostname), log_strdup(file_path));
+			err = fota_download_start(hostname, file_path);
+			if (err) {
+				LOG_ERR("Error when trying to start firmware"
+				       "download: %d", err);
+				return err;
+			}
+		} else if (execution_state == AWS_JOBS_IN_PROGRESS &&
+		    fota_state == APPLY_FIRMWARE) {
+			LOG_INF("Firmware download completed");
+			execution_state = AWS_JOBS_SUCCEEDED;
+			err = update_job_execution(client, job_id, execution_state,
+					fota_state, doc_version_number, "");
+			if (err) {
+				return err;
+			}
+		} else if (execution_state == AWS_JOBS_SUCCEEDED &&
+			   fota_state == APPLY_FIRMWARE) {
+			LOG_INF("Job document updated with SUCCEDED");
+			LOG_INF("Ready to reboot");
+			callback(AWS_FOTA_EVT_DONE);
+		}
+		return 1;
+	} else if (!strncmp(job_id_update_rejected_topic, topic,
+			    MIN(JOB_ID_UPDATE_TOPIC_MAX_LEN, topic_len))) {
+		LOG_ERR("Job document update was rejected");
```

Comment:
No payload expected here? If there's any, and not read, it'll break further processing (we're working with streams here). 

In case some payload was received, but you do not need to process it, you need to add a dummy read to flush it out from the socket.

Reply:
Ah okei, i'll add it back.

Reply:
Fixed